.github/workflows/ci-cd-pipeline.yml

```diff
@@ -0,0 +1,70 @@
+name: CI pipeline
+on:
+  push:
+    paths-ignore:
+      - README.md
+  pull_request:
+    paths-ignore:
+      - README.md
+jobs:
+  build:
+    name: Build project
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set environment variables
+        run: echo "${{ secrets.ENV_FILE }}" > .env
+      - name: build project
+        run: docker compose build production
+  pytest:
+    name: Run Pytest tests
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Run Pytest tests
+        run: docker compose --file docker-compose.pytest.yml up pytest --build
+  cypress:
+    name: Run Cypress tests
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set environment variables
+        run: echo "${{ secrets.ENV_FILE }}" > .env
+      - name: Run Cypress tests
+        run: docker compose up cypress --build --abort-on-container-exit
+  docker:
+    name: Publish to Docker Hub
+    needs:
+      - build
+      - pytest
+      - cypress
+    if: ${{ github.event_name == 'push' && github.ref_name == 'main' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set Docker Hub username environment variable
+        run: echo "${{ secrets.ENV_FILE }}" | grep "^DOCKERHUB_USERNAME=" >> $GITHUB_ENV
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ env.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Set environment variables
+        run: echo "${{ secrets.ENV_FILE }}" > .env
```

Comment:
Do you actually need this .env here? It's not used to build your images (I hope!) so I _think_ we could drop it here

Reply:
It totally was being used to build the images, have attempted to change that now!